asmcomp/cmmgen.ml

```diff
@@ -156,8 +156,7 @@ and mult_power2 c n dbg = lsl_int c (Cconst_int (Misc.log2 n)) dbg
 
 let rec mul_int c1 c2 dbg =
   match (c1, c2) with
-  | (_, Cconst_int 0) | (Cconst_int 0, _) ->
-      Cconst_int 0
+  | (c, Cconst_int 0) | (Cconst_int 0, c) -> Csequence (c, Cconst_int 0)
```

Comment:
I actually wrote it that way first, but it turns out to be a much bigger and more controversial change, and the `Csequence` style is [used elsewhere in this file](https://github.com/ocaml/ocaml/blob/545aaef50badf2aff6dabdac0ec35fb574e9adf2/asmcomp/cmmgen.ml#L342-L343).

I think it's better to fix this as simply and directly as possible, and consider `is_pure` as a separate enhancement.